### PR TITLE
Jonmv/multiple test and staging zones

### PR DESCRIPTION
**Pull request by jonmv**
@freva please review very carefully >_<

### CODE REVIEW

File: config-provisioning/src/main/java/com/yahoo/config/provision/zone/ZoneFilter.java [hunk 1]
```diff
@@ -16,24 +15,13 @@ public interface ZoneFilter {
     /** Negates the next filter. */
     ZoneFilter not();
 
-    /** Zones which are upgraded by the controller. */
```
Comment (freva) [inline config-provisioning/src/main/java/com/yahoo/config/provision/zone/ZoneFilter.java:19]: Remember to update implementation in hosted

**Review by freva — APPROVED** [review]

File: config-provisioning/src/main/java/com/yahoo/config/provision/zone/ZoneFilter.java (hunk 1 above)
Comment (jonmv) [reply config-provisioning/src/main/java/com/yahoo/config/provision/zone/ZoneFilter.java:19]: Yes :)

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentStatus.java [hunk 2]
```diff
@@ -538,11 +574,15 @@ public Map<JobId, List<Job>> testJobs(Map<JobId, List<Job>> jobs) {
...
                     }
-            });
-        }
+            }
+        });
         return Collections.unmodifiableMap(testJobs);
     }
 
+    private CloudName findCloud(JobType job) {
+        return zones.zones().all().among(job.zone()).zones().stream().findFirst().map(ZoneApi::getCloudName).orElse(null);
```
[the remark this answers is not in the document]
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentStatus.java:583]: Hmm, probably, yes ...

**Review by freva — APPROVED** [review]

**Comment by jonmv** [comment]
What I really don't like here is that mutiple `JobType` with the same identity (job name) can have different zones. I think it would be appropriate to split the identity and the zone in two different classes. However, the `.zone()` is used everywhere, so this will be a bit of a change ... 
The alternative, still, is to use multiple job names, and keep the zone 1-1 with name. Urgh :)

**Pull request merged**